aerogearcatalog/keycloak-apb
# [AEROGEAR-2564] Annotate the mobile client resource on bind, remove annotations on unbind

### Description
Annotates  the mobile client CRD with Keycloak specific details as part of the bind step. This allows us  to show deep links and other information for Keycloak within the context of a mobile client in the UI. All annotations should be deleted on unbind.

### Progress
- Annotate the mobile client via the client Id that is passed in during the bind
- Remove annotations on unbind
- Ensure that bind/unbind roles do not fail when client Id is incorrect/does not exist

### Verification Steps
1. Pull the latest master of the minishift mobile core addon and restart minishift
2. Successfully provision Keycloak from this branch and create a new mobile client
3. Create a new binding, passing in the correct client Id of the client created in step 2. Verify that the binding is successful and the client resource has been annotated with the correct values for `binding.aerogear.org/keycloak-uri-href` and `binding.aerogear.org/keycloak-realm`
4. Delete the binding. Verify that all resources created on bind have been cleaned up and that the mobile client resource is no longer annotated
5. Create a new binding with a non existent client Id. Verify that the bind and unbind steps do not fail.

### Related Jiras
[AEROGEAR-2564](https://issues.jboss.org/browse/AEROGEAR-2564)
[AEROGEAR-2640](https://issues.jboss.org/browse/AEROGEAR-2640)
